### PR TITLE
osd: bluestore: NVMEDevice: fix error handling

### PR DESCRIPTION
**Pull request by xiexingguo**
This pr contains two similar error handling cases and a clean up, which I hope might be helpful.

### CODE REVIEW

**Comment by xiexingguo** [comment]
@yuyuyu101 Do you mind having a look at? Thanks:-)

File: src/os/bluestore/NVMEDevice.cc [hunk 1]
```diff
@@ -504,6 +503,8 @@ int NVMEManager::try_get(const string &sn_tag, SharedDriverData **driver)
   ProbeContext ctx = {sn_tag, this, nullptr};
   r = spdk_nvme_probe(&ctx, probe_cb, attach_cb);
   if (r || !ctx.driver) {
+    if (!ctx.driver)
+      r = -ENOENT;
```
Comment (yuyuyu101) [inline src/os/bluestore/NVMEDevice.cc]: I don't think this is right. all errors will result in ctx.driver == nullptr, so we can't set r = -ENOENT indicate the specified error.

Comment (xiexingguo) [inline src/os/bluestore/NVMEDevice.cc]: @yuyuyu101 Well, I fully understand what you mean. But theoretically there is one possibility we hit the below condition:
```
  r == 0 && ctx.driver == nullptr
```
which will also bring we here.

So I think the right logic should be(based on your suggestion):
```
if (r < 0) {
  assert(!ctx.driver);
}
```

What do you think?

Comment (yuyuyu101) [inline src/os/bluestore/NVMEDevice.cc]: this should be ok

**Pull request merged**